### PR TITLE
added usable prefixes field to oLj command

### PR DESCRIPTION
**Pull request by deepakchethan**
For issue #12620

I will change the field name from `URIs` to something else if required.

The output for `oLj` is:

`{"IO_Plugins":[{"Permissions":"rw_","Name":"ar","Description":"Open ar/lib files [ar|lib]://[file//path]","License":"LGPL3","URIs":"[ar|lib]://file/path"},{"Permissions":"rw_","Name":"bfdbg","Description":"BrainFuck Debugger (bfdbg://path/to/file)","License":"LGPL3","URIs":"bfdbg://file/path"},{"Permissions":"rwd","Name":"bochs","Description":"Attach to a BOCHS debugger","License":"LGPL3","URIs":"bochs://file/path"},{"Permissions":"r_d","Name":"debug","Description":"Native debugger (dbg:///bin/ls dbg://1388 pidof:// waitfor://)","License":"LGPL3","URIs":"[dbg|pidof|waitfor|]://file/path","version":"0.2.0","plugin":"pancake"},{"Permissions":"rw_","Name":"default","Description":"open local files using def_mmap://","License":"LGPL3","URIs":"[file|nocache|]://file/path"},{"Permissions":"rwd","Name":"gdb","Description":"Attach to gdbserver, 'qemu -s', gdb://localhost:1234","License":"LGPL3","URIs":"gdb://localhost:1234"},{"Permissions":"rw_","Name":"gprobe","Description":"open gprobe connection using gprobe://","License":"LGPL3","URIs":"gprobe://file/path"},{"Permissions":"rw_","Name":"gzip","Description":"read/write gzipped files","License":"LGPL3","URIs":"gzip://file/path"},{"Permissions":"rw_","Name":"http","Description":"http get (http://rada.re/)","License":"LGPL3","URIs":"http://rada.re/"},{"Permissions":"rw_","Name":"ihex","Description":"Intel HEX file (ihex://eeproms.hex)","License":"LGPL","URIs":"ihex://file.hex"},{"Permissions":"rwd","Name":"mach","Description":"mach debugger io plugin (mach://pid)","License":"LGPL","URIs":"[attach|mach|smach]://pid"},{"Permissions":"rw_","Name":"malloc","Description":"memory allocation (malloc://1024 hex://cd8090)","License":"LGPL3","URIs":"[malloc|hex]://id"},{"Permissions":"rw_","Name":"mmap","Description":"open file using mmap://","License":"LGPL3","URIs":"mmap://id"},{"Permissions":"rw_","Name":"null","Description":"null-plugin (null://23)","License":"LGPL3","URIs":"null://id"},{"Permissions":"rwd","Name":"qnx","Description":"Attach to QNX pdebug instance, qnx://host:1234","License":"LGPL3","URIs":"qnx://[host]:[port]"},{"Permissions":"rw_","Name":"r2k","Description":"kernel access API io (r2k://)","License":"LGPL3","URIs":"r2k://"},{"Permissions":"rw_","Name":"r2pipe","Description":"r2pipe io plugin","License":"MIT","URIs":"r2pipe://pipe"},{"Permissions":"rw_","Name":"r2web","Description":"r2web io client (r2web://cloud.rada.re/cmd/)","License":"LGPL3","URIs":"r2web://host:ip/cmd/"},{"Permissions":"rw_","Name":"rap","Description":"remote binary protocol (rap://:port rap://host:port/file)","License":"MIT","URIs":"[rap|raps]://host:port/file"},{"Permissions":"rw_","Name":"rbuf","Description":"RBuffer IO plugin: rbuf://","License":"LGPL","URIs":"rbuf://path"},{"Permissions":"rw_","Name":"self","Description":"read memory from myself using 'self://'","License":"LGPL3","URIs":"self://ops"}`

### CODE REVIEW

**Comment by XVilka** [comment]
I think it makes sense to show them in JSON as an arrays.

File: libr/io/p/io_ihex.c [hunk 1]
```diff
@@ -436,7 +437,7 @@ RIOPlugin r_io_plugin_ihex = {
 #ifndef CORELIB
 R_API RLibStruct radare_plugin = {
 	.type = R_LIB_TYPE_IO,
-	.data = &r_io_plugin_hex,
+	.data = &r_io_plugin_ihex,
```
Comment (deepakchethan) [inline libr/io/p/io_ihex.c]: There was no `&r_io_plugin_hex`, it was giving me an error. So I changed this as well.

File: libr/io/p/io_ihex.c [hunk 2]
```diff
@@ -423,6 +423,7 @@ static bool __resize(RIO *io, RIODesc *fd, ut64 size) {
 RIOPlugin r_io_plugin_ihex = {
 	.name = "ihex",
         .desc = "Intel HEX file (ihex://eeproms.hex)",
+		.uris = "ihex://file.hex",
```
Comment (XVilka) [inline libr/io/p/io_ihex.c]: broken indent

Comment (deepakchethan) [reply libr/io/p/io_ihex.c]: I will fix the indentation on all the files.

**Comment by deepakchethan** [comment]
> I think it makes sense to show them in JSON as an arrays.

I updated with the `oLj` command now.

**Comment by deepakchethan** [comment]
I am adding internal fields and modifying a lot of files. I think I can come up with a better approach. I will close this PR and start working on a better approach.

File: libr/io/plugin.c [hunk 3]
```diff
@@ -112,6 +112,9 @@ R_API int r_io_plugin_list_json(RIO *io) {
 		io->cb_printf ("%s{\"Permissions\":\"%s\",\"Name\":\"%s\",\"Description\":\"%s\",\"License\":\"%s\"",
 				n? "," : "", str, plugin->name,
 			plugin->desc, plugin->license);
+		if (plugin->uris) {
```
Comment (Maijin) [inline libr/io/plugin.c]: Can you update this by using pj.c library instead?

**Comment by radare** [comment]
also keys in json shouldnt be capitalized

File: libr/io/plugin.c (hunk 3 above)
Comment (deepakchethan) [reply libr/io/plugin.c]: Sure, will use pc.j library. I will follow this up on this on #13013.

**Comment by deepakchethan** [comment]
I didn't modify the JSON keys actually. But I will definitely change them to lower case.

File: libr/io/plugin.c (hunk 3 above)
Comment (Maijin) [reply libr/io/plugin.c]: Don't close your PR next time, just push -f the changes, otherwise it's a mega-mess

Comment (deepakchethan) [reply libr/io/plugin.c]: Okay @Maijin, I will definitely keep that in mind. Thanks for the feedback.

I added the pj.c library to it and pushed.

**Pull request closed without merging**